rasa/server.py

```diff
@@ -476,6 +479,124 @@ async def hello(request: Request):
         return response.text("Hello from Rasa: " + rasa.__version__)
 
 
+def async_if_callback_url(f: Callable[..., Coroutine]) -> Callable:
+    """Decorator to enable async request handling.
+
+    If the incoming HTTP request specified a `callback_url` query parameter, the request
+    will return immediately with a 204 while the actual request response will
+    be sent to the `callback_url`. If an error happens, the error payload will also
+    be sent to the `callback_url`.
+
+    Args:
+        f: The request handler function which should be decorated.
+
+    Returns:
+        The decorated function.
+    """
+
+    @wraps(f)
+    async def decorated_function(
+        request: Request, *args: Any, **kwargs: Any
+    ) -> HTTPResponse:
+        callback_url = request.args.get("callback_url")
+        # Only process request asynchronously if the user specified a `callback_url`
+        # query parameter.
+        if not callback_url:
+            return await f(request, *args, **kwargs)
+
+        async def wrapped() -> None:
+            try:
+                result: HTTPResponse = await f(request, *args, **kwargs)
+                payload = dict(
+                    data=result.body, headers={"Content-Type": result.content_type}
```

Comment:
Any other headers we may want to forward?

Reply:
Mhm, everything else should be set automatically from scratch. We just need to encoding as only obtain the already encoded data. I also can't see other default from Sani when we call `response.json` than setting the content type